src/convertToObject.js

```diff
@@ -3,16 +3,34 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an example in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const trimmedString = sourceString
+    .split(';')
+    .filter(item => item.trim() !== '');
+
+  const propertyValuePairs = trimmedString.map(item => {
+    const [property, value] = item.split(':');
+
+    return [property.trim(), value.trim()];
+  });
+
+  const resultingObject = {};
+
+  propertyValuePairs.map(item => {
```

Comment:
- it doesn't make sense to use `map` if you don't use returned from its array
- You can use the reduce method to generate an object from the array. And actually, you can combine the first time you mapped through an array and this one and replace just with the reduce